provider/google.go

```diff
@@ -175,13 +175,19 @@ func (p *GoogleProvider) Records() (endpoints []*endpoint.Endpoint, _ error) {
 
 	f := func(resp *dns.ResourceRecordSetsListResponse) error {
 		for _, r := range resp.Rrsets {
-
+			if !supportedRecordType(r.Type) {
+				continue
+			}
+			ep := &endpoint.Endpoint{
+				DNSName:    strings.TrimSuffix(r.Name, "."),
+				RecordType: r.Type,
+				Targets:    make(endpoint.Targets, 0, len(r.Rrdatas)),
+			}
 			for _, rr := range r.Rrdatas {
 				// each page is processed sequentially, no need for a mutex here.
-				if supportedRecordType(r.Type) {
```

Comment:
why was this check dropped ?

Reply:
Because the check is now in the enclosing for loop. We need to check only once the type of the Resource Record and then I simply append all data in the Resource Record as targets.

Reply:
you are right, my bad :)